client/src/project_loading_notification_service.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import * as vscode from 'vscode';
+
+export class ProjectLoadingNotificationService {
+  private readonly options: vscode.ProgressOptions;
+  private task?: {resolve: () => void};
+
+  constructor() {
+    this.options = {
+      location: vscode.ProgressLocation.Window,
+      title: 'Initializing Angular language features',
+    };
+  }
+
+  start(): void {
+    this.reset();
+    vscode.window.withProgress(this.options, () => new Promise((resolve) => this.task = {resolve}));
+  }
+
+  finish(): void {
+    this.reset();
+  }
+
+  dispose(): void {
```

Comment:
Is this method ever used?

Reply:
yes, it will be used here: https://github.com/andrius-pra/vscode-ng-language-service/blob/0d810242c580a10907e5acc9d5f0dad36b73ccd5/client/src/extension.ts#L53-L57

Reply:
It seems doesn't work as I expected. I had disabled extension during project loading, but nothing changed. Loading status disappeared when project had loaded.
I removed this class.